### PR TITLE
fix: avoid direct access to reviews list AJAX

### LINKED ISSUE #37920: [Issue] Avoid direct access to reviews list Ajax
This issue is automatically created based on existing pull request: magento/magento2#33876: Avoid direct access to reviews list Ajax

---------
### Description (*)
Do not know wether it is intentional,
But you can access directly to the reviews list Ajax controller however all others controllers for Ajax URL has a conditional test in code.

Capture d’écran 2021-08-20 à 20 32 39

### Fixed Issues (if relevant)
Why not restrain this page to Ajax only

### Manual testing scenarios 
You can give it a try :
/review/product/listAjax/id/9999
=> Display the raw output in the browser.

(9999 = product if from store)

### Questions or comments

Give me your think,

Ilan Parmentier

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by Amadeco**
### Description (*)
Do not know wether it is intentional,
But you can access directly to the reviews list Ajax controller however all others controllers for Ajax URL has a conditional test in code.

Capture d’écran 2021-08-20 à 20 32 39

### Fixed Issues (if relevant)
Why not restrain this page to Ajax only

### Manual testing scenarios 
You can give it a try :
/review/product/listAjax/id/9999
=> Display the raw output in the browser.

(9999 = product if from store)

### Questions or comments

Give me your think,

Ilan Parmentier

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#37920: Avoid direct access to reviews list Ajax

### CODE REVIEW

**Comment by mageho** [comment]
@magento give me test instance

**Comment by mageho** [comment]
@magento run all tests

File: app/code/Magento/Review/Controller/Product/ListAjax.php [hunk 1]
```diff
@@ -22,7 +22,8 @@ class ListAjax extends ProductController implements HttpGetActionInterface
      */
     public function execute()
     {
-        if (!$this->initProduct()) {
+        if (!$this->getRequest()->isAjax()
```
Comment (bgorski) [inline app/code/Magento/Review/Controller/Product/ListAjax.php]: Please format your code according to PSR-12 guidelines:
> Expressions in parentheses MAY be split across multiple lines, where each subsequent line is indented at least once. When doing so, the first condition MUST be on the next line. 

So the code would become:
```
if (
    !$this->getRequest()->isAjax() ||
    !$this->initProduct()
) {
    // code here
}
````
Reference: Section 5.1 of https://www.php-fig.org/psr/psr-12/

**Review by bgorski — CHANGES_REQUESTED** [review]

**Comment by bgorski** [comment]
@mageho thank you for your contribution! Indeed, this action can be accessed directly and if indexed by search engines, it may contribute to bad user experience. I had one change request though - please address it and I think we're good to go.

File: app/code/Magento/Review/Controller/Product/ListAjax.php (hunk 1 above)
Comment (Amadeco) [reply app/code/Magento/Review/Controller/Product/ListAjax.php]: Alright, I just changed with your kind indications.

**Comment by mageho** [comment]
@bgorski Hello there, I did the modification by following your advice. We must be good to go. have a nice weekend.

**Review by bgorski — APPROVED** [review]
@mageho indeed we are - approving the PR. The next step is to wait for the core team to prioritize this and to perform manual testing. 
Thank you for your contribution!

**Comment by bgorski** [comment]
@magento run all tests

**Comment by magento-engcom-team** [comment]
Hi @bgorski, thank you for the review.
[ENGCOM-9191](https://jira.corp.magento.com/browse/ENGCOM-9191) has been created to process this Pull Request
:eight_spoked_asterisk: @bgorski, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by bgorski** [comment]
@magento run Unit Tests, Functional Tests CE, Functional Tests B2B

**Review by Den4ik — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, thank you for the review.
[ENGCOM-9191](https://jira.corp.magento.com/browse/ENGCOM-9191) has been created to process this Pull Request

**Comment by engcom-Lima** [comment]
@magento run all tests

**Comment by engcom-Lima** [comment]
@magento give me test instance

**Comment by engcom-Lima** [comment]
### :heavy_check_mark: QA Passed

**Preconditions:**
- Install fresh Magento **`2.4-develop`** and **`PHP 8.1`**

**Manual testing scenario:**
- Add `/review/product/listAjax/id/{productID}` after the instance frontend URL and hit.

**Before:** :heavy_multiplication_x: Showing the reviews list.
Screenshot 2023-08-25 at 3 53 14 PM

Screenshot 2023-08-25 at 3 46 39 PM

**After:** :heavy_check_mark:  Now no direct access to reviews list. 

Screenshot 2023-08-25 at 4 08 45 PM

Builds are failed. Hence, moving this PR to **Extended Testing.**

**Comment by engcom-Lima** [comment]
@magento create issue

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE, Static Tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests B2B, Unit Tests

**Comment by engcom-Charlie** [comment]
The functional B2B test failures are not consistent in recent 2 builds on same code. The failures are not part of PR or not failing because of PR changes, seems to be flaky. Hence moving it to Merge in Progress.

**Run 1:**
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/33876/cb4b10365d4ff3f9dbced57c13e3de5c/Functional/allure-report-b2b/index.html#categories/e874bb47202994ea819dd7cbe4bd2bbb/889461fc9b54c415/

image


**Run 2:** 
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/33876/f7ec58b11a8b881a673ce380a6a97aef/Functional/console-error-logs.html

image

**Comment by engcom-Dash** [comment]
@iparmentier Could you please sign the Adobe CLA so that we can proceed with this PR further.

Thanks!

**Comment by iparmentier** [comment]
@engcom-Dash Hello, considered it done !
